prov/sm2/src/sm2_comp.c

```diff
@@ -103,7 +104,7 @@ int sm2_complete_rx(struct sm2_ep *ep, void *context, uint32_t op,
 
 	cq = container_of(ep->util_ep.rx_cq, struct sm2_cq, util_cq);
 	return ep->rx_comp(cq, context, flags, len, buf,
-			   ep->region->map->peers[id].fiaddr, tag, data);
+			   id, tag, data);
```

Comment:
the original code is required to make AV_USER_ID correct, how does your change not break it?

Reply:
AV_USER_ID means that we write the peers addr (string) instead of their ID (int)?  In that case... we do break it... but everyone has the same ID as everyone else.